sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -296,8 +296,7 @@ private[hive] class HiveClientImpl(
       case e: NoClassDefFoundError
         if HiveUtils.isHive23 && e.getMessage.contains("org/apache/hadoop/hive/serde2/SerDe") =>
```

Comment:
@dongjoon-hyun should we remove `HiveUtils.isHive23` and related changes? I see it was added when we add Hive 2.3 support at https://github.com/apache/spark/commit/33f3c48cac087e079b9c7e342c2e58b16eaaa681#diff-842e3447fc453de26c706db1cac8f2c4R59. cc @wangyum FYI

Reply:
@AngersZhuuuu should we remove https://github.com/apache/spark/blob/55ce49ed28a58e0047a9680fa63f4dff54c11be4/sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationSuite.scala#L41 too?

Reply:
+1, we can remove `HiveUtils.isHive23` and related changes.